git-cliff-core/src/changelog.rs

```diff
@@ -144,6 +147,43 @@ impl<'a> Changelog<'a> {
 				})
 				.collect::<Vec<Commit>>();
 		});
+
+		if self.config.git.require_conventional.unwrap_or(false) {
+			self.check_conventional_commits()?;
+		}
+
+		Ok(())
+	}
+
+	/// Checks the commits and returns an error if any unconventional commits
+	/// are found.
+	fn check_conventional_commits(&mut self) -> Result<()> {
+		debug!("Verifying that all commits are conventional.");
+		let mut unconventional_count = 0;
+
+		self.releases.iter().for_each(|release| {
+			release.commits.iter().for_each(|commit| {
+				if commit.conv.is_none() {
+					error!(
+						"Commit {id} is not conventional:\n{message}",
+						id = &commit.id[..7],
+						message = commit
+							.message
+							.lines()
+							.map(|line| { format!("    | {}", line.trim()) })
+							.collect::<Vec<String>>()
+							.join("\n")
+					);
```

Comment:
How about we print in a single line when the commit is not multine?

```suggestion
					error!(
						"Commit {id} is not conventional: {message}",
						id = &commit.id[..7],
						message = {
							if commit.message.lines().count() == 1 {
								format!("{}", commit.message.trim())
							} else {
								commit
									.message
									.lines()
									.map(|line| format!("\n    | {}", line.trim()))
									.collect::<Vec<String>>()
									.join("")
							}
						}
					);
```

Results in:

```
 ERROR git_cliff_core::changelog > Commit 9cba393 is not conventional:
    | Merge pull request #7 from gitolith/feat/custom-error-handling
    |
    | Custom Errors
 ERROR git_cliff_core::changelog > Commit b5f0136 is not conventional: Merge branch 'master'
 ERROR git_cliff_core::changelog > Commit 223cfc9 is not conventional:
    | Merge pull request #9 from orhun/chore/fix-ci
    |
    | Fix CI
 ERROR git_cliff_core::changelog > Commit 74f605d is not conventional: Merge branch 'master' into feat/parse-commits
```

I find this better because we are not wasting newlines for commits that are single line.

What do you think?

Reply:
I prefer sticking to the multi-line approach. Changing the look of an error message because some messages are shorter than others makes it more difficult to parse at a glance.

Also I doubt many of our users pay by line, thus having a few more lines in the output won't hurt anyone ;)